sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessor.java

```diff
@@ -411,4 +324,277 @@ public void run() {
+
+    public static final class Builder {
+
+      private static final String KEY_SCHEDULE_DELAY_MILLIS = "otel.bsp.schedule.delay";
+      private static final String KEY_MAX_QUEUE_SIZE = "otel.bsp.max.queue";
+      private static final String KEY_MAX_EXPORT_BATCH_SIZE = "otel.bsp.max.export.batch";
+      private static final String KEY_EXPORT_TIMEOUT_MILLIS = "otel.bsp.export.timeout";
+      private static final String KEY_SAMPLED = "otel.bsp.report.sampled";
+
+      private static final long DEFAULT_SCHEDULE_DELAY_MILLIS = 5000;
+      private static final int DEFAULT_MAX_QUEUE_SIZE = 2048;
+      private static final int DEFAULT_MAX_EXPORT_BATCH_SIZE = 512;
+      private static final int DEFAULT_EXPORT_TIMEOUT_MILLIS = 30_000;
+      private static final boolean DEFAULT_REPORT_ONLY_SAMPLED = true;
+
+      private long scheduleDelayMillis = DEFAULT_SCHEDULE_DELAY_MILLIS;
+      private int maxQueueSize = DEFAULT_MAX_QUEUE_SIZE;
+      private int maxExportBatchSize = DEFAULT_MAX_EXPORT_BATCH_SIZE;
+      private int exporterTimeoutMillis = DEFAULT_EXPORT_TIMEOUT_MILLIS;
+      private boolean sampled = DEFAULT_REPORT_ONLY_SAMPLED;
+
+      enum NamingConvention {
+        DOT {
+          @Override
+          protected char getSeparator() {
+            return '.';
+          }
+
+          @Nonnull
+          @Override
+          public String normalize(@Nonnull String key) {
+            return key.toLowerCase();
+          }
+        },
+        ENV_VAR {
+          @Override
+          protected char getSeparator() {
+            return '_';
+          }
+        };
+
+        @Nonnull
+        public String normalize(@Nonnull String key) {
+          return key.toLowerCase().replace(getSeparator(), DOT.getSeparator());
+        }
+
+        public Map<String, String> normalize(@Nonnull Map<String, String> map) {
+          Map<String, String> properties = new HashMap<>();
+          for (String key : map.keySet()) {
+            properties.put(normalize(key), map.get(key));
+          }
+          return Collections.unmodifiableMap(properties);
+        }
+
+        protected abstract char getSeparator();
+      }
+
+      /**
+       * Sets the configuration values from the given configuration map. This method looks for the
+       * following keys:
+       *
+       * <ul>
+       *   <li>{@code otel.bsp.schedule.delay}: to set the delay interval between two consecutive
+       *       exports.
+       *   <li>{@code otel.bsp.max.queue}: to set the maximum queue size.
+       *   <li>{@code otel.bsp.max.export.batch}: to set the maximum batch size.
+       *   <li>{@code otel.bsp.export.timeout}: to set the maximum allowed time to export data.
+       *   <li>{@code otel.bsp.report.sampled}: to set whether only sampled spans should be
+       *       reported.
+       * </ul>
+       *
+       * @param configMap {@link Map} holding the configuration values.
+       * @return this.
+       */
+      public Builder fromConfigMap(
```

Comment:
Do we need to have this public?

Reply:
If someone wanted to use this to build config from a parsed yaml/json/xml file, this would be a handy entry point.

Reply:
Not really, you cannot extend the enum, so it will support only DOT or ENV for the enum. Also I feel like we should not expose helpers just in case someone needs it (we will suffer maintaining this API). I would suggest to remove it.

Reply:
I left it public because I thought that the agent might load the json/xml/yaml file and use this API to set the configuration object.
So the agent will properly create the map with the ENV or DOT naming convention.

Reply:
In the SIG call, we agreed on supporting Properties as an additional configuration medium. This method can be package visible only.